File: sdks/java/lib/src/main/java/models/shared/CollaborationAllowlistEntry.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * CollaborationAllowlistEntry
 * An entry that describes an approved domain for which users can collaborate
 * with files and folders in your enterprise or vice versa.
**/
public class CollaborationAllowlistEntry {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public CollaborationAllowlistEntry withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("direction")
    public CollaborationAllowlistEntryDirectionEnum direction;
    public CollaborationAllowlistEntry withDirection(CollaborationAllowlistEntryDirectionEnum direction) {
        this.direction = direction;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("domain")
    public String domain;
    public CollaborationAllowlistEntry withDomain(String domain) {
        this.domain = domain;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("enterprise")
    public CollaborationAllowlistEntryEnterprise enterprise;
    public CollaborationAllowlistEntry withEnterprise(CollaborationAllowlistEntryEnterprise enterprise) {
        this.enterprise = enterprise;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public CollaborationAllowlistEntry withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public CollaborationAllowlistEntryTypeEnum type;
    public CollaborationAllowlistEntry withType(CollaborationAllowlistEntryTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/collaborationallowlistentry.go
package shared

import (
	"time"
)

type CollaborationAllowlistEntryDirectionEnum string

const (
	CollaborationAllowlistEntryDirectionEnumInbound  CollaborationAllowlistEntryDirectionEnum = "inbound"
	CollaborationAllowlistEntryDirectionEnumOutbound CollaborationAllowlistEntryDirectionEnum = "outbound"
	CollaborationAllowlistEntryDirectionEnumBoth     CollaborationAllowlistEntryDirectionEnum = "both"
)

type CollaborationAllowlistEntryEnterpriseTypeEnum string

const (
	CollaborationAllowlistEntryEnterpriseTypeEnumEnterprise CollaborationAllowlistEntryEnterpriseTypeEnum = "enterprise"
)

// CollaborationAllowlistEntryEnterprise
// A representation of a Box enterprise
type CollaborationAllowlistEntryEnterprise struct {
	ID   *string                                        `json:"id,omitempty"`
	Name *string                                        `json:"name,omitempty"`
	Type *CollaborationAllowlistEntryEnterpriseTypeEnum `json:"type,omitempty"`
}

type CollaborationAllowlistEntryTypeEnum string

const (
	CollaborationAllowlistEntryTypeEnumCollaborationWhitelistEntry CollaborationAllowlistEntryTypeEnum = "collaboration_whitelist_entry"
)

// CollaborationAllowlistEntry
// An entry that describes an approved domain for which users can collaborate
// with files and folders in your enterprise or vice versa.
type CollaborationAllowlistEntry struct {
	CreatedAt  *time.Time                                `json:"created_at,omitempty"`
	Direction  *CollaborationAllowlistEntryDirectionEnum `json:"direction,omitempty"`
	Domain     *string                                   `json:"domain,omitempty"`
	Enterprise *CollaborationAllowlistEntryEnterprise    `json:"enterprise,omitempty"`
	ID         *string                                   `json:"id,omitempty"`
	Type       *CollaborationAllowlistEntryTypeEnum      `json:"type,omitempty"`
}
